lib/ecto/migration.ex

```diff
@@ -1096,14 +1096,16 @@ defmodule Ecto.Migration do
     struct(%Constraint{table: table, name: name}, opts)
   end
 
-  @doc """
-  Executes queue migration commands.
-
-  Reverses the order in which commands are executed when doing a rollback
-  on a `change/0` function and resets the commands queue.
-  """
-  def flush do
-    Runner.flush
+  @doc "Executes queue migration commands."
+  defmacro flush do
+    quote bind_quoted: [caller: __CALLER__.module] do
+      if direction() == :down and function_exported?(caller, :change, 0) do
```

Comment:
The migrator checks for down first and then falls back to `:change`, so we should express this here:

```suggestion
      if direction() == :down and not function_exported?(caller, :down, 0) do
```